xchart/src/main/java/org/knowm/xchart/internal/chartpart/AxisTickCalculator_Date.java

```diff
@@ -242,6 +242,9 @@ protected void calculate() {
         // }
       }
       //      System.out.println("************");
+      if(index >= timeSpans.size() - 1)
+        break; // We reached the end, even though we might not yet have the result we want,
+               // continuing will lead to an Exception!
     } while (skip
         || !areAllTickLabelsUnique(tickLabels)
         || !willLabelsFitInTickSpaceHint(tickLabels, gridStepInChartSpace));
```

Comment:
A tiny remark is that a break at the end of the loop can likely be replaced by changing the condition in `while`. Can you try that?

Reply:
The condition will get a lot more complicated and the explanation comment why this is necessary probably needs to be removed. Are you sure that's a good idea?

Reply:
I am not sure, but in general the readability of that loop is not perfect.

Reply:
Maybe we can just merge as is, and then clean-up the code afterwards.

Reply:
Do you have an idea when there will be a new release?

Reply:
I don't. AFAIK @timmolter is the only one who can issue a new release.